backend/editor/models.py

```diff
@@ -74,12 +75,12 @@ class EditHeaderParameters(CommonParameters):
 
 
 class EditHeaderResponse(BaseModel):
-    result: []
+    result: dict
 
 
 class EditFooterParameters(CommonParameters):
     incoming_data: Footer
 
 
 class EditFooterResponse(BaseModel):

```

Comment:
Hey @alexgarel 

I am getting the error:

```
Internal Server Error /openapi.json
```

The logs says:

```taxonomy-editor-taxonomy_frontend-1  | 172.22.0.1 - - [05/Feb/2023:13:05:24 +0000] "GET /docs HTTP/1.1" 200 959 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36" "-"
taxonomy-editor-taxonomy_api-1       | INFO:     172.22.0.1:0 - "GET /openapi.json HTTP/1.0" 500 Internal Server Error
taxonomy-editor-taxonomy_frontend-1  | 172.22.0.1 - - [05/Feb/2023:13:05:25 +0000] "GET /openapi.json HTTP/1.1" 500 21 "http://api.taxonomy.localhost:8091/docs" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/109.0.0.0 Safari/537.36" "-"
taxonomy-editor-taxonomy_api-1       | ERROR:    Exception in ASGI application
taxonomy-editor-taxonomy_api-1       | Traceback (most recent call last):
taxonomy-editor-taxonomy_api-1       |   File "/usr/local/lib/python3.9/site-packages/starlette/middleware/base.py", line 77, in call_next
taxonomy-editor-taxonomy_api-1       |     message = await recv_stream.receive()
taxonomy-editor-taxonomy_api-1       |   File "/usr/local/lib/python3.9/site-packages/anyio/streams/memory.py", line 94, in receive
taxonomy-editor-taxonomy_api-1       |     return self.receive_nowait()
taxonomy-editor-taxonomy_api-1       |   File "/usr/local/lib/python3.9/site-packages/anyio/streams/memory.py", line 87, in receive_nowait
taxonomy-editor-taxonomy_api-1       |     raise EndOfStream
taxonomy-editor-taxonomy_api-1       | anyio.EndOfStream

.
.
.
  File "/usr/local/lib/python3.9/site-packages/fastapi/applications.py", line 200, in openapi
taxonomy-editor-taxonomy_api-1       |     self.openapi_schema = get_openapi(
taxonomy-editor-taxonomy_api-1       |   File "/usr/local/lib/python3.9/site-packages/fastapi/openapi/utils.py", line 423, in get_openapi
taxonomy-editor-taxonomy_api-1       |     definitions = get_model_definitions(
taxonomy-editor-taxonomy_api-1       |   File "/usr/local/lib/python3.9/site-packages/fastapi/utils.py", line 45, in get_model_definitions
taxonomy-editor-taxonomy_api-1       |     m_schema, m_definitions, m_nested_models = model_process_schema(
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 590, in pydantic.schema.model_process_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 631, in pydantic.schema.model_type_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 258, in pydantic.schema.field_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 471, in pydantic.schema.field_type_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 858, in pydantic.schema.field_singleton_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 710, in pydantic.schema.field_singleton_sub_fields_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 536, in pydantic.schema.field_type_schema
taxonomy-editor-taxonomy_api-1       |   File "pydantic/schema.py", line 957, in pydantic.schema.field_singleton_schema
taxonomy-editor-taxonomy_api-1       | ValueError: Value not declarable with JSON Schema, field: name='_result' type=Entry required=True


```

[This is most relevant issue I could find while debugging.](https://github.com/tiangolo/fastapi/issues/3328)

To summarize it means I am returning wrong types from end points. 

For a few APIs there is a lot of nested function calls. Like `edit_header`, `edit_footer`. This article: https://json-schema.org/understanding-json-schema/reference/type.html share JSON equivalent type in Python. 

I have used `dict` for `edit_header`, `edit_footer` because I was not sure what kind of values they return.

**Could you confirm the return type in the response models is correct?**

Reply:
For the traceback, see my comment above.

We would like to avoid dict or list without any precision but we can iterate, and so if you put a `# TODO: document more` it's ok.